Objects/genobject.c

```diff
@@ -408,9 +408,16 @@ gen_close(PyGenObject *gen, PyObject *args)
         PyErr_SetString(PyExc_RuntimeError, msg);
         return NULL;
     }
-    if (PyErr_ExceptionMatches(PyExc_StopIteration)
-        || PyErr_ExceptionMatches(PyExc_GeneratorExit)) {
-        PyErr_Clear();          /* ignore these errors */
+    if (PyErr_ExceptionMatches(PyExc_StopIteration)) {
+        /* retrieve the StopIteration exception instance being handled, and
+         * extract its value */
+        PyObject *exc = PyErr_GetRaisedException();  /* clears the error indicator */
+        PyObject *value = Py_NewRef(((PyStopIterationObject *)exc)->value);
+        Py_DECREF(exc);
```

Comment:
There is actually _PyGen_FetchStopIterationValue already, which this logic is replicating.

Reply:
Do you prefer using that over the public API?

Reply:
I quite like it, the only downside I can see is that it clearing the error indicator is one step further removed from the immediate logic in `close()`.